pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -416,12 +439,102 @@ func newDeployment(etcdstorage *samplev1alpha1.EtcdStorage) *appsv1.Deployment {
+							Args:    []string{etcdaddr, etcdns, "--listen-addr=0.0.0.0:2379", "--cacert=/etc/certs/ca/ca.pem", "--cert=/etc/certs/client/client.pem", "--key=/etc/certs/client/client-key.pem"},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "etcd",
+									Protocol:      corev1.ProtocolTCP,
+									ContainerPort: 2379,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      c.coreEtcdOptions.CertSecretName,
+									MountPath: "/etc/certs/client",
+									ReadOnly:  true,
+								},
+								{
+									Name:      c.coreEtcdOptions.CAConfigMapName,
+									MountPath: "/etc/certs/ca",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: c.coreEtcdOptions.CertSecretName,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: c.coreEtcdOptions.CertSecretName,
+								},
+							},
+						},
+						{
+							Name: c.coreEtcdOptions.CAConfigMapName,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: c.coreEtcdOptions.CAConfigMapName,
+									},
+								},
+							},
 						},
 					},
 				},
 			},
 		},
 	}
 }
+
+func (c *EtcdProxyController) newService(etcdstorage *samplev1alpha1.EtcdStorage) *corev1.Service {
+	labels := map[string]string{
+		"controller": "epc",
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("etcd-%s", etcdstorage.Name),
+			Namespace: c.controllerNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(etcdstorage, schema.GroupVersionKind{
+					Group:   samplev1alpha1.SchemeGroupVersion.Group,
+					Version: samplev1alpha1.SchemeGroupVersion.Version,
+					Kind:    "EtcdStorage",
+				}),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol:   corev1.ProtocolTCP,
+					Port:       2379,
+					TargetPort: intstr.FromInt(2379),
+				},
+			},
+		},
+	}
+}
+
+// GetControllerNamespace returns name of the namespace where controller is located. The namespace name is obtained
+// from the "/var/run/secrets/kubernetes.io/serviceaccount/namespace" file. In case it's not possible to obtain it
+// from that file, the function resorts to the default name, `kube-apiserver-storage`.
+func GetControllerNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
+			return ns
+		}
+	}
+
+	runtime.HandleError(fmt.Errorf("unable to detect controller namespace"))
+	return "kube-apiserver-storage"
+}
+
+// getReplicaSetName calculates name to be used to create a ReplicaSet.
+func getReplicaSetName(etcdstorage *samplev1alpha1.EtcdStorage) string {
```

Comment:
follow-up: avoid "get" prefixes. Just rename this to etcdstorageReplicaSetName. No need for an ugly get prefix.

Reply:
Already done. Just ended up in different PR when I split them earlier today.